images/cfw/opt/x1plus/lib/python/x1plus/utils.py

```diff
@@ -7,12 +7,22 @@
 def is_emulating():
     return not os.path.exists("/etc/bblap")
 
-
 @lru_cache(None)
 def serial_number():
     """
     Used to get the Serial Number for the Printer
     """
-    return subprocess.check_output(["bbl_3dpsn"], stderr=subprocess.DEVNULL).decode(
-        "utf-8"
-    )
+    if os.path.isfile("/oem/device/sn"):
+        return open("/oem/device/sn", "r").read()
+    else:
+        if os.system("bbl_3dpsn 2>/dev/null > /tmp/.bambu_sn") != 0:
+            return open("/tmp/.bambu_sn", "r").read() 
```

Comment:
Wait why are we doing this bonkers `/tmp/.bambu_sn` thing again?  That is the whole point of the `subprocess.check_output(["bbl_3dpsn"])`, so we odn't have to create any other files.  And every supported X1Plus installation should be able to run `bbl_3dpsn`, there should be no need for `/oem/device/sn` at this point.

Reply:
Oh ok, I was just copy pasting what I saw used in another one of our Python scripts (although I have now forgotten which one).